rct229/ruleset_functions/baseline_systems/is_baseline_system_9b.py

```diff
@@ -0,0 +1,73 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.are_all_terminal_heat_sources_hot_water import (
+    are_all_terminal_heat_sources_hot_water,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.are_all_terminal_heating_loops_purchased_heating import (
+    are_all_terminal_heating_loops_purchased_heating,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.are_all_terminal_types_CAV_with_none_equal_to_null import (
+    are_all_terminal_types_cav_with_none_equal_to_null,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.do_all_terminals_have_one_fan import (
+    do_all_terminals_have_one_fan,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.does_each_zone_have_only_one_terminal import (
+    does_each_zone_have_only_one_terminal,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_cooling_type_none import (
+    is_hvac_sys_cooling_type_none,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    HVAC_SYS,
+    has_fan_system,
+    has_heating_system,
+    has_preheat_system,
+)
+
+HEATING_SYSTEM = schema_enums["HeatingSystemOptions"]
+
+
+def is_baseline_system_9b(rmi_b, hvac_b_id, terminal_unit_id_list, zone_id_list):
+    """
+    Returns true or false to whether the baseline system type is 9b (system 9 with purchased HW).
+
+    Parameters
+     ----------
+     rmi_b json
+         To evaluate if the hvac system is modeled as either Sys-9bin the B_RMI.
+     hvac_b_id list
+         The id of the hvac system to evaluate.
+     terminal_unit_id_list
+         List list of terminal unit IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+     zone_id_list list
+         list of zone IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    Returns
+     -------
+     Returns true or false to whether the baseline system type is 9b (system 9 with purchased HW).
+    """
+
+    is_baseline_system_9b = HVAC_SYS.UNMATCHED
+
+    # if preheat, heating, and fan systems DON'T exist, has_required_sys=True, else, False
+    has_required_sys = not (
+        has_preheat_system(rmi_b, hvac_b_id)
+        and has_heating_system(rmi_b, hvac_b_id)
+        and has_fan_system(rmi_b, hvac_b_id)
+    )
+
+    return (
+        # short-circuit the logic if no required data is found.
+        has_required_sys
+        # sub functions handles missing required sys, and return False.
+        and is_hvac_sys_cooling_type_none(rmi_b, hvac_b_id)
```

Comment:
@weilixu  Could we use `has_cooling_system` instead of `is_hvac_sys_cooling_type_none`?  I am wondering if the intension is the same. I know that the implementation is slightly different.

Reply:
It could be difficult since `is_hvac_sys_cooling_type_none` is written in RDS. The other way around could be a lot easier.

I do see the intention is a bit different. `has_cooling_system` is checking whether there is a cooling system in an HVAC system, and `is_hvac_sys_cooling_type_none` is checking whether the cooling type in HVAC is None or Null. So these two functions are opposite to each other.

One thing we could do is to revise the `has_cooling_system` to:
```python
def has_cooling_system(rmi, hvac_id):
    return not is_hvac_sys_cooling_type_none(rmi, hvac_id)
```

Reply:
@weilixu @charliepnnl 
This is a great point! However, When I implemented the updated ```has_cooling_system``` function, as Weili described, there is an error.

This might be because ```is_hvac_sys_cooling_type_none``` function imports the ```find_exactly_one_hvac_system``` function in the ```baseline_system_util.py```, but the ```find_exactly_one_hvac_system```  function is in the```baseline_system_util.py``` and tries to import the ```is_hvac_sys_cooling_type_none``` function. So, two files try to import the functions in each file.

So, could we leave it as is to prevent the error?

Reply:
I am OK with the approach.

Reply:
Thank you Weili! I'll merge this PR.